### PR TITLE
Reduce interrupt iterator checks & field access

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

The interrupt iterator currently introduces a non-trivial amount of overhead to queries by checking for interrupts every 256 points. This commit adjusts that check to every 5000 points.

There are also several places where nested field access has been adjusted to minimize field lookups.

###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: influxql/iterator.gen.go [hunk 1]
```diff
@@ -4558,11 +4558,11 @@ func (itr *stringInterruptIterator) Stats() IteratorStats { return itr.input.Sta
...
 
 func (itr *stringInterruptIterator) Next() (*StringPoint, error) {
-	// Only check if the channel is closed every 256 points. This
-	// intentionally checks on both 0 and 256 so that if the iterator
+	// Only check if the channel is closed every N points. This
+	// intentionally checks on both 0 and N so that if the iterator
 	// has been interrupted before the first point is emitted it will
 	// not emit any points.
-	if itr.count&0x100 == 0 {
+	if itr.count&5000 == 0 {
```
Comment (mark-rushakoff) [inline influxql/iterator.gen.go]: Both the current and previous implementation are checking too frequently. The comparison should not be 0, but rather the same value as the bitmask.

IOW, [x&5000 == 0 is true for 256 values from 0-5000](https://play.golang.org/p/TjMHRXYB8M), but [x&5000 == 5000 is only true once from 0-5000](https://play.golang.org/p/N9Vp_TW7vs).

Comment (benbjohnson) [inline influxql/iterator.gen.go]: Ah, good catch. Weird. Not sure how I missed that. :facepalm:

Comment (benbjohnson) [inline influxql/iterator.gen.go]: Fixed in 9cdfcdd.

**Comment by toddboom** [comment]
@benbjohnson After talking to @mark-rushakoff, I think this still isn't quite right.

I think the bitmask needs to be `0xFF (255)` or `0xFFF (4095)` - something where all the bottom bits are `1`, and checked for equality against itself otherwise you'll keep matching once the bits roll over. So, the line should be more like:

```
itr.count&0xFF == 0xFF
```

Let me know if that seems crazy.

**Comment by benbjohnson** [comment]
@toddboom @mark-rushakoff Ah, you're totally right. Fixed in 1b6524a7bf65742241332cfe775a712858ad1db1. Bits are hard.

**Comment by toddboom** [comment]
binary math is at least 2^n times harder than regular math. lgtm. :shipit:

**Comment by benbjohnson** [comment]
@toddboom I think you meant, _"binary math is at least 1<<n times harder than regular math"_ :)

**Comment by benbjohnson** [comment]
@toddboom Actually, I guess they're both binary math. nm. :)

**Pull request merged**